packages/fe-fpm-writer/src/page/common.ts

```diff
@@ -0,0 +1,136 @@
+import type { Editor } from 'mem-fs-editor';
+import { join } from 'path';
+import type { ManifestNamespace } from '@sap-ux/ui5-config';
+import { validateBasePath } from '../common/validate';
+import type {
+    CustomPage,
+    FCL,
+    InternalCustomPage,
+    InternalObjectPage,
+    ObjectPage,
+    Navigation,
+    InternalListReport
+} from './types';
+import type { Manifest } from '../common/types';
+import { FCL_ROUTER } from '../common/defaults';
+
+/**
+ * Add a new route to the provided route array, and update existing routes if necessary (e.g. if targets are defined as arrays for FCL).
+ *
+ * @param routes existing application routes (from the manifest)
+ * @param config configuration object
+ */
+export function updateRoutes(
+    routes: ManifestNamespace.Route[],
+    config: InternalCustomPage | InternalObjectPage | InternalListReport
+) {
+    const newRoute: ManifestNamespace.Route = {
+        name: `${config.entity}${config.name}`
```

Comment:
what if the new route already exists?

Reply:
Then a 2nd route with the same name is added to the `routes` array. The resulting app still works but ignores the previously defined route.
For me that is fine for the writer module because it still generates valid content. The responsibility to check first is with the caller.

Reply:
OK